wiki/Tutorials/Elixir_OIDC/Elixir_AAI_OIDC.md

```diff
@@ -0,0 +1,231 @@
+The *oidc.crypto_passphrase* can be freely chosen.
+
+```
+# public keystone url
+<VirtualHost  XX.XX.XX.XX:5000>
+	...
+	ServerName https://{{controller.public_hostname}}:5000
+
+	OIDCResponseType "code"
+	OIDCClaimPrefix "OIDC-"
+	OIDCClaimDelimiter ;
+	OIDCScope "openid"
+	OIDCProviderMetadataURL {{ oidc.provider_url }}.well-known/openid-configuration
+	OIDCClientID {{ oidc.client_id }}
+	OIDCClientSecret {{ oidc.client_secret }}
+	OIDCCryptoPassphrase {{ oidc.crypto_passphrase }}
+	OIDCRedirectURI https://{{ controller.public_hostname }}:5000/identity/v3/auth/OS-FEDERATION/websso/oidc/redirect
+
+	# OAuth for CLI access
+	OIDCOAuthIntrospectionEndpoint {{ oidc.provider_url }}introspect
+	OIDCOAuthClientID {{ oidc.client_id }}
+	OIDCOAuthClientSecret {{ oidc.client_secret }}
+
+	<Location ~ "/v3/auth/OS-FEDERATION/websso/oidc">
+	     AuthType  openid-connect
+	     Require   valid-user
+	</Location>
+
+	<Location ~ "/v3/OS-FEDERATION/identity_providers/elixir_oidc/protocols/oidc/auth">
+	     Authtype oauth20
+	     Require   valid-user
+	</Location>
+```
+
+### Keystone configuration
+Configure your keystone.conf to include in the [auth] section oidc in the list of authentication methods and the keystone.auth.plugins.mapped.Mapped class for its implementation: 
+
+```
+[auth]
+methods = password,token,oidc
+oidc = keystone.auth.plugins.mapped.Mapped
+```
+
+Add your horizon host as trusted dashboard to the [federation] section and configure the SSO callback template.
+```
+[federation]
+trusted_dashboard=https://{{ oidc.endpoint }}/horizon/auth/websso/
+sso_callback_template = /etc/keystone/sso_callback_template.html
+```
+
+Define the identity provider in the [oidc] section using the HTTP_OIDC_ISS attribute
+```
+[oidc]
+remote_id_attribute = HTTP_OIDC_ISS
+```
+
+
+## Dashboard (Horizon)
+
+After setup Keystone to be OIDC aware, we have to a enable SSO (single sign on) in Openstack horizon and add Elixir_OIDC as choice in the list of providers via mapped drivers.
+
+```
+WEBSSO_ENABLED = True
+WEBSSO_CHOICES = (
+  ("credentials", _("Keystone Credentials")),
+  ("oidc", _("ELIXIR_OIDC")),
+)
+
+WEBSSO_INITIAL_CHOICE="oidc"
+```
+
+### Running Dashboard behind a HA-Proxy
+If you run Openstack in a HA setup behind a HA-Proxy you have to make sure that you configure the keystone public url and that this url can be reached by all Openstack controller. Otherwise the redirect from Elixir back to Keystone will not work.
+
+```
+OPENSTACK_HOST = "{{ controller.public_hostname }}"
+OPENSTACK_KEYSTONE_URL = "https://%s:5000/v3" % OPENSTACK_HOST
+OPENSTACK_ENDPOINT_TYPE = "publicURL"
+```
+
+### Running Dashboard behind a Proxy
+
+If your controller uses a http[s] proxy to connect to external addresses you have to configure the apache to make use of it. The environment can be set in `/etc/apache2/envvars`. Don't forget to configure `no_proxy` if your proxy only proxies external addresses.
```

Comment:
Be sure to restart the Webserver with `service apache2 restart`. A reload won't load in any changes made to the envvars file.